test/test_quantization.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+import torch.nn.quantized as nnq
+import torch.quantization as tq
+from torch.quantization import *
+from common_utils import TestCase, run_tests
+
+class SingleLayerLinearModel(torch.nn.Module):
+    def __init__(self):
+        super(SingleLayerLinearModel, self).__init__()
+        self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        return x
+
+class TwoLayerLinearModel(torch.nn.Module):
+    def __init__(self):
+        super(TwoLayerLinearModel, self).__init__()
+        self.fc1 = torch.nn.Linear(5, 8).to(dtype=torch.float)
+        self.fc2 = torch.nn.Linear(8, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.fc2(x)
+        return x
+
+class LinearReluModel(torch.nn.Module):
+    def __init__(self):
+        super(LinearReluModel, self).__init__()
+        self.fc = torch.nn.Linear(5, 5).to(dtype=torch.float)
+        self.relu = torch.nn.ReLU()
+
+    def forward(self, x):
+        x = self.relu(self.fc(x))
+        return x
+
+class NestedModel(torch.nn.Module):
+    def __init__(self):
+        super(NestedModel, self).__init__()
+        self.sub1 = LinearReluModel()
+        self.sub2 = TwoLayerLinearModel()
+        self.fc3 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+
+    def forward(self, x):
+        x = self.sub1(x)
+        x = self.sub2(x)
+        x = self.fc3(x)
+        return x
+
+class ManualQuantModel(torch.nn.Module):
+    def __init__(self):
+        super(ManualQuantModel, self).__init__()
+        self.quant = QuantStub(default_qconfig)
```

Comment:
if we treat QuantStub same as other modules - should we not receive qconfig here? (or make it optional)

Reply:
in our tool when we insert QuantStub the qconfig propagation is finished, so we need to provide a qconfig. I can make it optional, when qconfig is not provided, QuantStub will not have a `qconfig` field so that propagate_qconfig can go through.